Rubberduck.Inspections/Concrete/UnreachableCaseInspection.cs

```diff
@@ -0,0 +1,1309 @@
+                    }
+                }
+                selectStmtDO.CaseClauseDOs[idx] = caseClauseDO;
+            }
+            return selectStmtDO;
+        }
+
+        private SelectStmtDataObject InspectSelectStmtCaseClauses(SelectStmtDataObject selectStmtDO)
+        {
+            for (var idx = 0; idx < selectStmtDO.CaseClauseDOs.Count(); idx++)
+            {
+                var caseClause = selectStmtDO.CaseClauseDOs[idx];
+
+                if (selectStmtDO.SummaryCaseClauses.CaseElseIsUnreachable)
+                {
+                    caseClause.ResultType = ClauseEvaluationResult.Unreachable;
+                    selectStmtDO.CaseClauseDOs[idx] = caseClause;
+                    continue;
+                }
+
+                caseClause.ResultType = caseClause.RangeClauseDOs.All(rg => selectStmtDO.SummaryCaseClauses.RangeClausesAsText.Contains(rg.AsText)) ?
+                    ClauseEvaluationResult.Unreachable : caseClause.ResultType;
+
+                if (caseClause.ResultType == ClauseEvaluationResult.NoResult)
+                {
+                    caseClause = InspectCaseClause(caseClause, selectStmtDO.SummaryCaseClauses);
+                    selectStmtDO = UpdateSummaryClauses(selectStmtDO, caseClause);
+
+                    if (caseClause.RangeClauseDOs.All(rg => rg.ResultType != ClauseEvaluationResult.NoResult))
+                    {
+                        caseClause.ResultType = caseClause.RangeClauseDOs.All(rg => rg.ResultType == ClauseEvaluationResult.Unreachable)
+                            ? ClauseEvaluationResult.Unreachable : caseClause.RangeClauseDOs.All(rg => rg.ResultType == ClauseEvaluationResult.MismatchType)
+                                ? ClauseEvaluationResult.MismatchType : ClauseEvaluationResult.NoResult;
+                    }
+                }
+                selectStmtDO.CaseClauseDOs[idx] = caseClause;
+            }
+            return selectStmtDO;
+        }
+
+        private CaseClauseDataObject CreateCaseClauseDataObject(CaseClauseContext ctxt, string targetTypeName)
+        {
+            var caseClauseDO = new CaseClauseDataObject(ctxt);
+            var rangeClauseContexts = ParserRuleContextHelper.GetChildren<RangeClauseContext>(ctxt);
+            foreach (var rangeClauseCtxt in rangeClauseContexts)
+            {
+                var rgC = new RangeClauseDataObject(rangeClauseCtxt, targetTypeName);
+                caseClauseDO.RangeClauseDOs.Add(rgC);
+            }
+            return caseClauseDO;
+        }
+
+        private RangeClauseDataObject InitializeRangeClauseDataObject(RangeClauseDataObject rangeClauseDO, string targetTypeName, string refName )
+        {
+            rangeClauseDO.TypeNameTarget = targetTypeName;
+            rangeClauseDO.IdReferenceName = refName;
+            rangeClauseDO.UsesIsClause = HasChildToken(rangeClauseDO.Context, Tokens.Is);
+            rangeClauseDO.IsValueRange = HasChildToken(rangeClauseDO.Context, Tokens.To);
```

Comment:
HasChildToken should also be an extension method

Reply:
Added extension method(s).  If the intent is to refactor ParserRuleContextHelper completely into extension methods, then I would like to make that a separate Issue/PR.  I left ParserRuleContextHelper as-is.